More robustification for mixture model tests

lindsayad opened the pull request:
Follow-on to https://github.com/idaholab/moose/pull/26658

lindsayad committed:
Remove WCNSFV2PSlipVelocityAux

If a user still wants aux behavior they can use `FunctorAux` with the
functor material

Refs #22637 #26227

lindsayad committed:
Make sure we're getting same calcs for residual and Jacobian

`WCNSFV2PMomentumAdvectionSlip` was overriding the residual but
not the Jacobian (nor the gathering of RC data for that matter).
This was leading to poor nonlinear convergence in the channel test

Refs #22637 #26227

lindsayad committed:
Since not dropping derivs, we need ghosting for slip calculations

Also incorporate Peter's doc string grammar fix

Refs #22637 #26227

lindsayad committed:
Switch from aux kernels to functor materials in two phase inputs

Refs #22637 #26227

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/2033570/) on 1ebcb7a wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/26683/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2033614/) on 1ebcb7a wanted to post the following:


## Framework coverage

Coverage did not change


## Modules coverage

### Navier stokes

<table>
  <thead>
    <tr>
      <th></th>
      <th>41d8b3</th>
      <th colspan=4>#26683 1ebcb7</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>84.73%</td>
      <td>84.77%</td>
      <td>+0.04%</td>
      <td>89.58%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>14564</td>
      <td>14567</td>
      <td>+3</td>
      <td>86</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>2625</td>
      <td>2618</td>
      <td>-7</td>
      <td>10</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/26683/coverage_diff/navier_stokes)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/26683/coverage/navier_stokes)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/26683/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/26683/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/26683/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/26683/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/26683/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/26683/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/26683/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/26683/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/26683/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/26683/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/26683/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/26683/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/26683/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/26683/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/26683/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/26683/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/26683/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/26683/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/26683/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/26683/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/26683/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/26683/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/26683/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/26683/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/26683/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/26683/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/26683/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/26683/coverage/xfem)

</details>

## Warnings

- `navier_stokes` new line coverage rate 89.58% is less than the suggested 90.0%



This comment will be updated on new commits.

lindsayad committed:
Throw exception when about take non-integer power of negative number

In this case we're in pretty bad shape because the thing that is
negative is the Reynolds number, which is definitely unphysical

Refs #22637 #26227

GiudGiud commented on modules/navier_stokes/test/tests/finite_volume/two_phase/mixture_model/channel-advection-slip.i:
```diff
@@ -226,10 +237,12 @@ velocity_interp_method = 'rc'
     particle_diameter = ${dp}
     linear_coef_name = 'Darcy_coefficient'
+    outputs = 'out'
+    output_properties = 'vel_slip_x'
```
noice

lindsayad replied:
haha I was excited to use your new capability!

GiudGiud reviewed (approved):
looks good.

tanoret reviewed (approved):
All good. Nice changes!

We will need the AuxKernels at some point for the linear solvers, but we can add them back later.

lindsayad commented:
failures are unrelated
